komga-webui/src/functions/book-spreads.ts

```diff
@@ -9,40 +9,36 @@ export function buildSpreads(pages: PageDtoWithUrl[], pageLayout: PagedReaderLay
   if (pageLayout !== PagedReaderLayout.SINGLE_PAGE) {
     const spreads = [] as PageDtoWithUrl[][]
     const pagesClone = cloneDeep(pages)
-    let lastPages = undefined
     if (pageLayout === PagedReaderLayout.DOUBLE_PAGES) {
       const firstPage = pagesClone.shift() as PageDtoWithUrl
       if (isPageLandscape(firstPage))
         spreads.push([firstPage] as PageDtoWithUrl[])
       else
         spreads.push([createEmptyPage(firstPage), firstPage] as PageDtoWithUrl[])
-      if (pagesClone.length > 0) {
-        const lastPage = pagesClone.pop() as PageDtoWithUrl
-        if(isPageLandscape(lastPage))
-          lastPages = [lastPage] as PageDtoWithUrl[]
-        else
-          lastPages = [lastPage, createEmptyPage(lastPage)] as PageDtoWithUrl[]
-      }
     }
     while (pagesClone.length > 0) {
       const p = pagesClone.shift() as PageDtoWithUrl
       if (isPageLandscape(p)) {
         spreads.push([p])
       } else {
         if (pagesClone.length > 0) {
-          const p2 = pagesClone.shift() as PageDtoWithUrl
-          if (isPageLandscape(p2)) {
-            spreads.push([p, createEmptyPage(p)])
-            spreads.push([p2])
+          const [prevPage] = spreads.length > 0 ? (spreads[spreads.length - 1] as PageDtoWithUrl[]) : []
+          if (pageLayout === PagedReaderLayout.DOUBLE_NO_COVER && isPageLandscape(prevPage)){
+            spreads.push([createEmptyPage(p), p])
           } else {
-            spreads.push([p, p2])
+            const p2 = pagesClone.shift() as PageDtoWithUrl
+            if (isPageLandscape(p2)) {
+              spreads.push([p, createEmptyPage(p)])
+              spreads.push([p2])
+            } else {
+              spreads.push([p, p2])
+            }
```

Comment:
Basically all the same logic, just if it's DBNC layout, then just push a empty page after the previous page if and only if it was a landscape page. This guarantees the layouts will be opposite if landscape pages existed, which is what throws off the flow.